v3/src/models/document/free-tile-row.ts

```diff
@@ -131,9 +136,11 @@ export const FreeTileRow = TileRowModel
       self.maxZIndex = zIndex
     },
     insertTile(tileId: string, options?: ITileInRowOptions) {
-      const { x = 50, y = 50, width = undefined, height = undefined, zIndex = this.nextZIndex() } =
-        isFreeTileInRowOptions(options) ? options : {}
+      const {
+        x = 50, y = 50, width = undefined, height = undefined, zIndex = this.nextZIndex(), animateCreation = false
+      } = isFreeTileInRowOptions(options) ? options : {}
       self.tiles.set(tileId, { tileId, x, y, width, height, zIndex })
+      animateCreation && self.animateCreationTiles.add(tileId)
     },
     removeTile(tileId: string) {
       self.tiles.delete(tileId)
```

Comment:
Is it worth removing this tileId from animateCreationTiles?

Reply:
No, because that would prevent animation on redo of creation. Essentially, the logic here is that if a tile was user-created from the tool shelf, it should animate any time its component mounts for the lifetime of this browser session. In practice, the only times that the tile mounts are when it's created initially and when the creation is re-done, but the fact that it is a tile that should be animated on creation is a persistent property within the session, not one that's tied to whether the tile is currently in the document.